chain/db/Delegation.go

```diff
@@ -0,0 +1,17 @@
+package db
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func (b *BandDB) AddDelegation(
```

Comment:
Please also handle `MsgDelegate` to make sure we keep track of delegations correctly.